script_facade/models/r1/medication_order.py

```diff
@@ -44,7 +47,45 @@ def from_xml(cls, xml_element):
             }
         }
 
-        med_order = cls(med_cc, date_written=date_written, date_ended=None)
+        quantity_dispensed = xml_element.xpath('.//*[local-name()="Quantity"]//*[local-name()="Value"]')[0].text
+        dispense_request = {}
+        if quantity_dispensed:
+            dispense_request = {
+                'quantity': {
+                    'value': int(quantity_dispensed)
+                }
+            }
+
+        # todo: move these extensions to a separate MedicationDispense resource
+        pharmacy_name = xml_element.xpath('.//*[local-name()="Pharmacy"]//*[local-name()="StoreName"]')[0].text
+        if pharmacy_name:
+            dispense_request.setdefault('extension', [])
+            dispense_request['extension'].append(
+                {
+                    'url': 'http://cosri.org/fhir/pharmacy_name',
+                    'valueString': pharmacy_name,
+                }
+            )
+
+        last_fill = xml_element.xpath('.//*[local-name()="LastFillDate"]//*[local-name()="Date"]')[0].text
+        if last_fill:
+            dispense_request.setdefault('extension', [])
+            dispense_request['extension'].append(
+                {
+                    'url': 'http://cosri.org/fhir/last_fill',
+                    'valueDate': last_fill,
+                }
+            )
+
+        prescriber_fname = xml_element.xpath('.//*[local-name()="Prescriber"]//*[local-name()="Name"]//*[local-name()="FirstName"]')[0].text
+        prescriber_lname = xml_element.xpath('.//*[local-name()="Prescriber"]//*[local-name()="Name"]//*[local-name()="LastName"]')[0].text
+
+        # use contained resource, or save for other resource relationships?
+        prescriber = {
+            "display": " ".join((prescriber_fname, prescriber_lname))
+        }
+
+        med_order = cls(med_cc, dispense_request=dispense_request, prescriber=prescriber, date_written=date_written, date_ended=None)
```

Comment:
you may consider this a stylistic choice, but i don't see the win of having to extend the __init__ method on each additional attribute.  why not create the empty `med_order()` and assign each attribute as it's found?

Reply:
Thanks for the tip! I'm refactoring/cleaning up and the suggestions are much appreciated